### PR TITLE
support serialization for core::array::Range

### PR DESCRIPTION
**Pull request by tgolsson**
For the second issue from #718.  This is what the graph looks like:

image

This ends up failing with:

```
    Caused by:
        0: Serializing tensor "Range_1523_end.dat": ,TDim N
        1: Undetermined symbol in expression: N
```

I know that the [ONNX spec for Range](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Range) says the start,end,step are scalars, so I'm wondering if rewriting Range to do the same would be prudent? It doesn't seem like I can easily convert the Tensor to another RValue for serialization.

### CODE REVIEW

**Comment by kali** [comment]
I think I used the Tensor for the three Range in order to be type agnostic over the actual type of three scalars. As a matter of fact the Range code https://github.com/sonos/tract/blob/main/core/src/ops/array/range.rs#L106 make enough assumptions already about these values being scalar that I think it is safe to proceed like it (we could add an ensure!() in output_fact() though).

Then we have the eternal type snafu. ONNX accepts float or ints for Range. In tract, I assume these ints can be dynamic dimensions so I make a special case for TDim late resolution. NNEF has its own quirk: it uses "Scalar" to designate the tensor payload (which are actualy float or quantized values), and "Integer" are actually shape dimensions that should usually be mapped to TDim.

So... do we need two ranges in NNEF ? One expecting TDim, one expecting floats ? I dont think so. Because if the Range is float based, it will have been resolved by the constant folding happening in decluttering. The only thing that can prevent constant folding here is the presence of a dynamic dimension.

Bottom line, we can make the assumption that the three parameters are TDim, or integer castable to TDim (one of them at least being a "true" TDim). And they are Scalar in the tract sense, single value. 

So. Parameters type in NNEF should actually be `Integer.named("...")`. We can use `cast_to_scalar::<TDim>()` to extract the three values, and use `tdim()` to serialize them inline. And no more need for external tensor storage.

**Comment by tgolsson** [comment]
Thanks @kali! So if I'm following correctly: change Range to contain three TDims, and then rewrite serialization code based on that?

**Comment by kali** [comment]
Don't change Range in tract_core, we need it to handle the float case. Just assume you'll find single valeus TDim in ser/deser.

**Review by kali — APPROVED** [review]
